### ISSUE TITLE
nata-list.icu

### ISSUE THREAD

**Issue by NanoMeow** [NanoMeow/QuickReports#1258]
### Basic Information

**Test link:** `http://nata-list.icu/cg/`
**Category:** other
**Reported from:** Australia
**Template version:** 2

### User Environment

**Browser:** Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.38 Safari/537.36
**Extension:** Nano Adblocker 1.0.0.110

### Additional Message

NSFW

ads, (scam) popup, anti-adblock etc. in these links / domains: 

```
https://imgview.net/6aij2vbe6xct/1361616omT.jpg.html
https://imgview.net/cap380g9xy9q/1361617tbz.jpg.html

https://imgmaze.com/ztouhky7qqlj/1344713CaE.jpg.html
https://imgmaze.com/c97v5t0te0oe/1344714Yen.jpg.html

https://imgoutlet.com/5hieb7ekbzq6/2269768MBg.jpg.html
https://imgoutlet.com/nbm1mc2d31m9/2269769lbN.jpg.html

https://imgrock.net/uew0dj05itb8/5406937nYS.jpg.html
https://imgrock.net/48s9x6iu5fjf/5533988cDS.jpg.html

https://imgdew.com/ug7o2at8sioz/7155072fjO.jpg.html
https://imgdew.com/yhsiu4tzpoqr/7155074IXe.jpg.html

http://imgblaze.net/gw6ccqri8
http://imgblaze.net/wm8hqavj2

http://imgfrost.net/dkajckh
http://imgfrost.net/dgwiu
```

**Comment by jspenguin2017** [comment]
@okiehsch Eh... This is going to be a lot of pain...

**Comment by okiehsch** [comment]
Which links are problematic on your end?

**Comment by okiehsch** [comment]
On my end I can't connect to the last two domains and the first one has no issue.

**Comment by okiehsch** [comment]
I only fixed issues I could reproduce, so if someone is able to connect to the last two domains there could be more filters needed.

**Comment by jspenguin2017** [comment]
Can you connect to this one? `http://www.imgfile.net/dkajckh`

**Comment by okiehsch** [comment]
Yes, but  
![image](https://user-images.githubusercontent.com/16838044/57992545-0f8bcc00-7ab5-11e9-8e84-946023fbd1c9.png)

**Comment by okiehsch** [comment]
I don't feel like spending much time fixing a blocked site.

**Comment by jspenguin2017** [comment]
This works: 
```
imgfile.net##+js(window.open-defuser.js)
imgfile.net##div:has(:scope > .ssdvv)
imgfile.net##.odlake, .qwrcb
```

**Comment by okiehsch** [comment]
OK, I will add it.

**Comment by okiehsch** [comment]
Fixed, thanks.

**Comment by jspenguin2017** [comment]
@okiehsch Looks like there's still a few issues: 

```
# Popup / redirection
https://imgmaze.com/ztouhky7qqlj/1344713CaE.jpg.html
https://imgmaze.com/c97v5t0te0oe/1344714Yen.jpg.html

# NSFW ads
https://imgrock.net/uew0dj05itb8/5406937nYS.jpg.html
https://imgrock.net/48s9x6iu5fjf/5533988cDS.jpg.html
```

**Comment by jspenguin2017** [comment]
From comments from @LiCybora , the redirection only happens when some scripts fail to load: https://github.com/NanoAdblocker/NanoFilters/issues/333#issuecomment-493832189

**Comment by jspenguin2017** [comment]
Also annoyance on `imgdew.pw`, but that's pretty minor.

**Comment by okiehsch** [comment]
I can fix the ads, the redirection is rather tricky without breaking the continue button.

**Comment by jspenguin2017** [comment]
It's not like the continue button is usable right now.

**Comment by okiehsch** [comment]
Fixed the ads.

**Comment by jspenguin2017** [comment]
Oh these obfuscated scripts... 

I see 3 options: 
- Block ads & redirection, it's not like the button is usable right now
- Whitelist the scripts, we can worry about the unblocked popups later
- Block the entire domain

Looks like we tried to fix this multiple times, and they seem to be watching us, this might going be sticky...

**Comment by okiehsch** [comment]
Even if I whitelist every request I get redirected at 
`https://imgmaze.com/ztouhky7qqlj/1344713CaE.jpg.html`

**Comment by jspenguin2017** [comment]
Hum... It does work in guest mode though, although there's 3 popunders.

**Comment by okiehsch** [comment]
Wait, I whitelisted every request but if I block every request and remove all ##js filters there is no redirection on my end.

**Comment by okiehsch** [comment]
Does it work for you adding
````
imgmaze.*#@#+js(popads-dummy.js)
imgmaze.*#@#+js(abort-on-property-read.js, CustomEvent)
||imgmaze.*^$script,1p
````
to your list?

**Comment by jspenguin2017** [comment]
Nope... Does it work for you?

**Comment by okiehsch** [comment]
Well, yes.

**Comment by jspenguin2017** [comment]
That's weird... It still redirects for me. 
![image](https://user-images.githubusercontent.com/7283682/57997523-5c0de200-7a8a-11e9-864b-cfcd9610393c.png)

**Comment by ghajini** [comment]
Still have problem on `imgmaze.com` and `imgrock.net`
1. On imgmaze, I get redirected to always to continue to image page even if I click on it....
This is required then `@||imgmaze.pw/mcndhek.js$script,first-party`

2. On `imgrock.pw`,  it doesn't get to destinations page, always redirected  to `amiderations.info` shady site...
All is related to that `.js` seen in logger

**Comment by okiehsch** [comment]
It work for me with both sites after 5-7 blocked popunders, so I think this is the best possible fix for the moment that I can see.
I would strongly suggest to never use those sites anyway.

**Comment by jspenguin2017** [comment]
Most of the problems are resolved, opening a new issue for the remaining difficult problems.
